### PR TITLE
Add unit tests for macros

### PR DESCRIPTION
**Pull request by rlh1994**
## Description & motivation
The goal of this PR is to add a suite of basic unit tests for the macros in the package to ensure they correctly return expected SQL for the various warehouses. This adds a new argument to the functions that exists simply to avoid having to call the `is_run_with_new_events` macros and allows me to delay adding seeds and proper on-db integration tests to the package (which will be required for bigquery colum combination).

The focus of this was giving dummy inputs that are easier to debug, rather than test the full processing of full snowplow data.

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have raised a [documentation](https://github.com/snowplow/documentation) PR if applicable (N/A)

### CODE REVIEW

File: integration_tests/macros/test_snakeify_case.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    {% set expected_dict = {
+        "single_word" : "hello",
+        "already_snake" : "hello_world",
+        "pascal_case" : "hello_world",
+        "camel_basic" : "hello_world",
+        "camel_multi_word" : "hello_world_earth",
+        "camel_end_with_caps" :"hello_world_earth",
+        "camel_middle_caps" : "hello_world_earth",
+        "camel_hypthons" : "hello_world_earth",
```
Comment (emielver) [inline integration_tests/macros/test_snakeify_case.sql:34]: Here again

File: integration_tests/macros/test_snakeify_case.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    {% set results_dict ={
+        "single_word" : snowplow_normalize.snakeify_case('hello'),
+        "already_snake" : snowplow_normalize.snakeify_case('hello_world'),
+        "pascal_case" : snowplow_normalize.snakeify_case('HelloWorld'),
+        "camel_basic" : snowplow_normalize.snakeify_case('helloWorld'),
+        "camel_multi_word" : snowplow_normalize.snakeify_case('helloWorldEarth'),
+        "camel_end_with_caps" : snowplow_normalize.snakeify_case('helloWorldEARTH'),
+        "camel_middle_caps" : snowplow_normalize.snakeify_case('helloWORLDEarth'),
+        "camel_middle_caps" : snowplow_normalize.snakeify_case('helloWORLDEarth'),
+        "camel_hypthons" : snowplow_normalize.snakeify_case('helloWorld-Earth'),
```
Comment (emielver) [inline integration_tests/macros/test_snakeify_case.sql:47]: Here as well

File: integration_tests/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# snowplow-normalize-integration-tests
+
+Integration test suite for the snowplow-normalize dbt package.
+
+The `./scripts` directory contains one scripts:
+
+- `integration_tests.sh`: This tests the macros of the snowplow-normalize package. It physically checks the output of the macros to ensure it matches what we expect, but doesn't actually process any data directly.
```
Comment (emielver) [inline integration_tests/README.md:7]: I'd remove physically here, seems a bit out of place

File: integration_tests/macros/test_snakeify_case.sql [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+It runs 9 tests:
+1) a single world, no change
+2) already snake case, no change
+3) pascal case
+4) camel case, basic
+5) camel case with multiple workds
+6) camel case ending with a series of capitals
+7) camel case with a series of capitals in the middle
+8) camel case with hypthons
```
Comment (emielver) [inline integration_tests/macros/test_snakeify_case.sql:12]: Should be *hyphen I think?

**Review by emielver — APPROVED** [review]
Not sure I like the variable name `test` but it won't be used by the user and I wouldn't know what else to name it without making it unnecessarily verbose. Also spotted some typos but once those are cleaned up you're all good to go!

File: integration_tests/macros/test_snakeify_case.sql (hunk 4 above)
Comment (rlh1994) [reply integration_tests/macros/test_snakeify_case.sql:12]: I looked at this word and said to myself it didn't look right, but then just went with it anyway. Too much thinking about python...

**Pull request merged**